packages/bpk-component-segmented-control/src/BpkSegmentedControl.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ReactNode} from 'react';
+import { useState } from 'react';
+
+import { cssModules } from '../../bpk-react-utils';
+
+import SEGMENT_TYPES from './segmentTypes';
+
+import type { SegmentTypes } from './segmentTypes';
+
+import STYLES from './BpkSegmentedControl.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  buttonContents: string[] |  ReactNode[];
+  type?: SegmentTypes;
+  onItemClick: (id: number) => void; // index param to track which one is clicked
+  selectedIndex: number;
+  shadow?: boolean;
+};
+
+const BpkSegmentedControl = ({
+  buttonContents,
+  onItemClick,
+  selectedIndex,
+  shadow,
+  type = SEGMENT_TYPES.CanvasDefault,
+}: Props) => {
+  const [selectedButton, setSelectedButton] = useState(selectedIndex);
```

Comment:
Is selectedIndex every changed on the consumer side or is it just used for setting the initial state, if the former I would expect state handling and setting of values to be on the consumer side and Backpack components would be stateless

Reply:
We're using selectedIndex initially as a default value for displaying which button is selected on render. Then the value does change based on user input. Only one can be selected at a time, so having boolean prop of `selected` on each button does not work as an option. We had a look at `BpkSelectableChip` as an example, but I do not think this works.

Reply:
What I am thinking here, is that this logic could be on the consumer side as part of your onclick where you set the `selectedIndex` which is passed through to the component and then used as the value to do the logic on L59 - L60,

That way we don't have to do the `handleButtonClick` and mix both the state management and then the click on the consumer side.

As some folks may also see the API as `selectedIndex` is something they need to update as each button is clicked. Then you only need to pass the `onItemClick` and pass straight to the button :)

Reply:
resolving, per slack conversation we cant access index outside the component